components/content/content-editor/plugins/input-rules/detect-mention-suggestion.ts

```diff
@@ -0,0 +1,26 @@
+import { InputRule } from 'prosemirror-inputrules';
+import { EditorState } from 'prosemirror-state';
+import AppContentEditor from '../../content-editor';
+import { ContentEditorService } from '../../content-editor.service';
+import { ContentEditorSchema } from '../../schemas/content-editor-schema';
+
+export function detecctMentionSuggestionRule(editor: AppContentEditor) {
+	return new InputRule(
+		/@([\w_-]+)$/i,
+		(
+			state: EditorState<ContentEditorSchema>,
+			_match: string[],
+			_start: number,
+			_end: number
+		) => {
+			// We don't want to insert lists inside code text.
+			if (ContentEditorService.checkCurrentNodeIsCode(state)) {
+				return null;
+			}
+
+			editor.canShowMentionSuggestions = 2;
```

Comment:
why 2? seems like this is variable is some weird after thought in its current state haha